File: app/src/main/java/seigneur/gauvain/mycourt/utils/RoomConverter.java
package seigneur.gauvain.mycourt.utils;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.Date;
import java.util.Map;

public class RoomConverter {

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }


    @TypeConverter
    public static String stringArrayToString(String[] arrayString) {
        Gson gson = new Gson();
        return arrayString == null ? null : gson.toJson(arrayString);
    }

    @TypeConverter
    public static String[] stringToArrayString(String string) {
        Type listType = new TypeToken<String[]>() {}.getType();
       return new Gson().fromJson(string, listType);
    }

    @TypeConverter
    public static ArrayList<String> arrayListFromString(String value) {
        Type listType = new TypeToken<ArrayList<String>>() {}.getType();
        Gson gson = new Gson();
        return gson.fromJson(value, listType);
    }

    @TypeConverter
    public static String stringFromArrayList(ArrayList<String> list) {
        Gson gson = new Gson();
        return list == null ? null : gson.toJson(list);
    }

    @TypeConverter
    public static Map<String, String> stringToMapString(String string) {
        Type listType = new TypeToken<Map<String, String>>() {}.getType();
        return new Gson().fromJson(string, listType);
    }

    @TypeConverter
    public static String mapStringToString(Map<String, String> mapString) {
        Gson gson = new Gson();
        return mapString == null ? null : gson.toJson(mapString);
    }

   /* @TypeConverter
    public static mapToString(Map<String, String>) {
        return value == null ? null : Gson.toJson(value);
    }

    @TypeConverter
    public static Map<NexoIdentifier, Date> fromMap(String value) {
        return value == null ? null : Gson.fromJson(value, ...);
    }*/



}

File: app/src/main/java/seigneur/gauvain/mycourt/utils/RoomConverter.kt
package seigneur.gauvain.mycourt.utils

import androidx.room.TypeConverter

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import seigneur.gauvain.mycourt.data.model.Attachment

import java.lang.reflect.Type
import java.util.ArrayList
import java.util.Date


object RoomConverter {

    @TypeConverter
    @JvmStatic
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    @JvmStatic
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }


    @TypeConverter
    @JvmStatic
    fun stringArrayToString(arrayString: Array<String>?): String? {
        val gson = Gson()
        return if (arrayString == null) null else gson.toJson(arrayString)
    }

    @TypeConverter
    @JvmStatic
    fun stringToArrayString(string: String): Array<String>? {
        val listType = object : TypeToken<Array<String>>() {

        }.type
        return Gson().fromJson<Array<String>>(string, listType)
    }

    @TypeConverter
    @JvmStatic
    fun arrayListFromString(value: String): ArrayList<String>? {
        val listType = object : TypeToken<ArrayList<String>>() {

        }.type
        val gson = Gson()
        return gson.fromJson<ArrayList<String>>(value, listType)
    }

    @TypeConverter
    @JvmStatic
    fun stringFromArrayList(list: ArrayList<String>?): String? {
        val gson = Gson()
        return if (list == null) null else gson.toJson(list)
    }

    @TypeConverter
    @JvmStatic
    fun stringToMapString(string: String?): Map<String, String>? {
        val listType = object : TypeToken<Map<String, String>>() {}.type
        return Gson().fromJson<Map<String, String>>(string, listType)
    }

    @TypeConverter
    @JvmStatic
    fun mapStringToString(mapString: Map<String, String>?): String? {
        val gson = Gson()
        return if (mapString == null) null else gson.toJson(mapString)
    }

    @TypeConverter
    @JvmStatic
    fun attachmentListToJson(value: List<Attachment>?): String {
        return Gson().toJson(value)
    }

    @TypeConverter
    @JvmStatic
    fun attachmentjsonToList(value: String): List<Attachment>? {
        val objects = Gson().fromJson(value, Array<Attachment>::class.java) as Array<Attachment>
        val list = objects.toList()
        return list
    }

    @TypeConverter
    @JvmStatic
    fun intArrayToString(array: IntArray?): String? {
        val gson = Gson()
        return if (array == null) null else gson.toJson(array)
    }

    @TypeConverter
    @JvmStatic
    fun stringToArrayIntArray(string: String): IntArray? {
        val listType = object : TypeToken<IntArray>() {

        }.type
        return Gson().fromJson<IntArray>(string, listType)
    }



}